### PR TITLE
fix: remove metadata in descr in dtype_decode function. it fix gh-1152

### PR DESCRIPTION
**Pull request by ninousf**
Remove metadata in dtype_decode function. It fixes gh-1152, because h5py puts metadata in dtypes which are not correctly managed in numpy

### CODE REVIEW

**Comment by joshmoore** [comment]
Thanks for this, @ninousf. I've kicked off the workflows. A heads up that there look to be quite a few (unintended?) whitespace changes.

Also a question: do you think it would help to have access to h5py for testing?

**Comment by ninousf** [comment]
sorry, the spaces are because i use yapf. 
h5py can be useful for tests, i think there are many cases where scientists use both formats and convert h5 to zarr and vice-versa

**Review by joshmoore — CHANGES_REQUESTED** [review]
> sorry, the spaces are because i use yapf.

Can you revert them or open a new PR to simplify review?

> h5py can be useful for tests, i think there are many cases where scientists use both formats and convert h5 to zarr and vice-versa

:+1: If you want to try it in this PR, feel free by adding it to https://github.com/zarr-developers/zarr-python/blob/main/requirements_dev_optional.txt; otherwise we can look into this as a follow-up.

File: zarr/meta.py [hunk 1]
```diff
@@ -187,18 +190,27 @@ def encode_dtype(cls, d: np.dtype):
...
         if isinstance(d, list):
             # recurse to handle nested structures
-            d = [(k[0], cls._decode_dtype_descr(k[1])) + tuple(k[2:]) for k in d]
+            d = [(k[0], cls._decode_dtype_descr(k[1])) + tuple(k[2:])
+                 for k in d]
         return d
 
     @classmethod
     def decode_dtype(cls, d) -> np.dtype:
+        print('before', d)
```
[the remark this answers is not in the document]
Comment (joshmoore) [reply zarr/meta.py:206]: Happy to, but independently I think we should get this green to know that it's stable beforehand, no?

**Pull request closed without merging**